src/dune_cache/shared.ml

```diff
@@ -65,28 +65,27 @@ struct
     | Error exn -> Miss (Error (Printexc.to_string exn))
   ;;
 
-  let lookup_impl ~rule_digest ~targets =
+  let lookup_impl ~can_go_in_shared_cache ~rule_digest ~targets =
     match config with
     | Disabled -> Fiber.return (Hit_or_miss.Miss Miss_reason.Cache_disabled)
     | Enabled { storage_mode = mode; reproducibility_check } ->
-      (match Config.Reproducibility_check.sample reproducibility_check with
-       | true ->
-         (* CR-someday amokhov: Here we re-execute the rule, as in Jenga. To make
-            [check_probability] more meaningful, we could first make sure that
-            the shared cache actually does contain an entry for [rule_digest]. *)
-         Fiber.return (Hit_or_miss.Miss Miss_reason.Rerunning_for_reproducibility_check)
-       | false -> try_to_restore_from_shared_cache ~mode ~rule_digest ~targets)
+      if can_go_in_shared_cache
+      then (
+        match Config.Reproducibility_check.sample reproducibility_check with
+        | true ->
+          (* CR-someday amokhov: Here we re-execute the rule, as in Jenga. To make
+             [check_probability] more meaningful, we could first make sure that
+             the shared cache actually does contain an entry for [rule_digest]. *)
+          Fiber.return (Hit_or_miss.Miss Miss_reason.Rerunning_for_reproducibility_check)
+        | false -> try_to_restore_from_shared_cache ~mode ~rule_digest ~targets)
+      else Fiber.return (Hit_or_miss.Miss Miss_reason.Cannot_go_in_shared_cache)
   ;;
 
   let lookup ~can_go_in_shared_cache ~rule_digest ~targets
     : Digest.t Targets.Produced.t option Fiber.t
     =
     let open Fiber.O in
-    let+ result =
-      match can_go_in_shared_cache with
-      | false -> Fiber.return (Hit_or_miss.Miss Miss_reason.Cannot_go_in_shared_cache)
```

Comment:
What does this change do? As far as I can tell, it just makes the function return `Cache_disabled` rather than `Cannot_go_in_shared_cache`. Does that have an impact?

Reply:
You might have missed this comment? Don't think the changes in this file are necessary.

Reply:
I think the code used to read whether to override the value in the config, but I agree that with the current state this seems to not be necessary. I've reverted it and will rebase and squash it later.